lib/gemstash/gem_yanker.rb

```diff
@@ -0,0 +1,58 @@
+require "gemstash"
+
+module Gemstash
+  # Class that supports yanking a gem from the private repository of gems.
+  class GemYanker
+    include Gemstash::Env::Helper
+
+    # This error is thrown when yanking a non-existing gem name.
+    class UnknownGemError < StandardError
+    end
+
+    # This error is thrown when yanking a non-existing gem version.
+    class UnknownVersionError < StandardError
+    end
+
+    # This error is thrown when yanking an already yanked gem version.
+    class YankedVersionError < StandardError
+    end
+
+    def initialize(auth_key, gem_name, slug)
+      @auth_key = auth_key
+      @gem_name = gem_name
+      @slug = slug
+      @db_helper = Gemstash::DBHelper.new
+    end
+
+    def yank
+      check_auth
+      update_database
+      invalidate_cache
+    end
+
+  private
+
+    def check_auth
+      raise Gemstash::NotAuthorizedError, "Authorization key required" if @auth_key.to_s.strip.empty?
+      auth = Authorization[@auth_key]
+      raise Gemstash::NotAuthorizedError, "Authorization key is invalid" unless auth
+      raise Gemstash::NotAuthorizedError, "Authorization key doesn't have yank access" unless auth.yank?
```

Comment:
I find the Authorization[key] a bit strange, I think it is because it is sitting the middle of the method in between raises.

Does it make sense to turn it into that looks something like Authorization.can?(key, :yank) ?

I guess in my head it makes sense to make it symmetric with the Authorization.authorize(key, %w(yank unyank)) that we are using for the opposite use.